### ISSUE TITLE
GUI - ClickHouse Database - Error with a boolean field 

### ISSUE THREAD

**Issue by eden0907** [dbeaver/dbeaver#16157]
#### System information: 
- Windows 11 - 64 bits Professional
- Dbeaver CE Version : 22.0.2.202204041542

#### Connection specification:
- ClickHouse version :  ClickHouse server version 22.3.3 revision 54455.
- jdbc driver : Yandex ClickHouse driver (server version 21.3+)
- NO SSH, just HTTP

**Very Strange problem** 
I have created a table with a Nullable(Boolean) field.
when I do a select * from this table : OK data is will printed
but If I add a where  condition, Dbeaver print a message "Unsupported Data field Bool"
I tried with the clickhouse client utility, and it works fine, I got data correctly without any message .

In DBeaver, I tried with creating an another test table, inserting, getting and all works fine ?!@ 

**Backtrace :** 
org.jkiss.dbeaver.model.exec.DBCException: Unsupported data type: Bool
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:852)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:3515)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:118)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:116)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:4854)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.jkiss.dbeaver.DBException: Unsupported data type: Bool
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:230)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:458)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:847)
	... 7 more
Caused by: java.lang.IllegalArgumentException: Unsupported data type: Bool
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:1055)
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:882)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$MappedFunctions.deserialize(ClickHouseRowBinaryProcessor.java:459)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.readNextRow(ClickHouseRowBinaryProcessor.java:521)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:551)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:496)
	at com.clickhouse.jdbc.ClickHouseResultSet.next(ClickHouseResultSet.java:716)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.next(JDBCResultSetImpl.java:272)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:180)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.fetchQueryData(SQLQueryJob.java:751)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:575)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:451)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	... 9 more

![2022-04-12 09_06_52-DBeaver 22 0 2 - _SCM ClickHouse_Script-262](https://user-images.githubusercontent.com/103486731/162915581-00d66356-40f1-420a-87a1-7534cff16c31.png)
![2022-04-12 09_09_47-DBeaver 22 0 2 - _SCM ClickHouse _ Script-262](https://user-images.githubusercontent.com/103486731/162915582-e669afcf-f960-479d-b097-18ccd6064706.png)

**Comment by LonwoLonwo** [comment]
Hello @eden0907 

Can you provide your table DDL, please?

**Comment by eden0907** [comment]
Hello,   

This is the DDL table script 
and some insert data.  continuing analyzing the problem,  I found that some detailled lines works fine  ( scenarioid = 2 in my example) but not with scenarioid = 2454.   If mixing a "good" line and the "bad" line, I got a good result. It's very strange.


CREATE TABLE scenario
(
 `scenarioid` Int32,
 `versioncode` Int32,
 `label` Nullable(String),
 `ispublished` Nullable(Bool),   -- I got the same Error with declaring Nullable(Boolean)
 `scenariotypecode` Nullable(Int32),
 `userid` Nullable(Int32),
 `nomenclaturecode` Nullable(Int32),
 `bustructurecode` Nullable(Int32),
 `buarray` Array(Int32),
 `bannercomparray` Array(Int32),
 `bucomparray` Array(Int32),
 `dashboardid` Nullable(Int32),
 `tagarray` Array(Int32),
 `creationdate` Nullable(DateTime),
 `modificationdate` Nullable(DateTime),
 `suppressiondate` Nullable(DateTime),
 `nbdayvaliditysurvey1` Nullable(Int32),
 `nbdayvaliditysurvey2` Nullable(Int32),
 `scenarioproductfiltertypecode` Nullable(Int32),
 `elementnomenclaturearray` Array(Int32),
 `analysisaxisid` Nullable(Int32),
 `analysisaxiselementarray` Array(Int32),
 `externaldatasourcetypecode` Nullable(Int32),
 `marketindexid` Nullable(Int32),
 `sharedscenariosourceid` Nullable(Int32),
 `shortcomment` Nullable(String),
 `issalesperiodautomatic` Nullable(Bool),   -- I got the same Error with declaring Nullable(Boolean)
 `salesperiodcode` Nullable(String),
 `startdate` Nullable(DateTime),
 `enddate` Nullable(DateTime),
 `priority` Nullable(Int16),
 `budgetid` Nullable(Int32),
 `layoutid` Nullable(Int32)
)
ENGINE = MergeTree
PRIMARY KEY (scenarioid, versioncode)
ORDER BY (scenarioid, versioncode)
SETTINGS allow_nullable_key = 0,
 index_granularity = 8192;


INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2, 1, 'First', true, 2, 4, 1, 3, [1], [], [], 4, [], '2020-10-30 16:34:56', '2020-10-30 17:05:37', NULL, 45, NULL, 1, [133], NULL, [], NULL, 2, NULL, NULL, false, 'Qty4S', NULL, NULL, NULL, NULL, NULL);

INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2454, 1, 'Sugg. Cde Magasins', true, 45, 102, 1, 3, [286,291,294,299,303,1000], [], [], NULL, [], '2022-03-31 10:19:12', '2022-03-31 10:19:12', NULL, NULL, NULL, NULL, [17], NULL, [], NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 410);

INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2454, 2, 'Sugg. Cde Magasins', true, 45, 102, 1, 3, [299,295,304], [], [], NULL, [], '2022-03-31 14:33:26', '2022-03-31 14:54:59', NULL, NULL, NULL, NULL, [17], NULL, [], NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 410);

INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2454, 3, 'Sugg. Cde Magasins', true, 45, 102, 1, 3, [299,295,304], [], [], NULL, [], '2022-03-31 14:57:56', '2022-03-31 14:58:47', NULL, NULL, NULL, NULL, [17], NULL, [], NULL, NULL, NULL, 'Essai', NULL, NULL, NULL, NULL, NULL, NULL, 410);

INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2454, 4, 'Sugg. Cde Magasins', true, 45, 102, 1, 3, [299,295,304], [], [], NULL, [], '2022-03-31 15:05:59', '2022-03-31 17:48:19', NULL, NULL, NULL, NULL, [17], NULL, [], NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 410);

INSERT INTO scenario
(scenarioid, versioncode, label, ispublished, scenariotypecode, userid, nomenclaturecode, bustructurecode, buarray, bannercomparray, bucomparray, dashboardid, tagarray, creationdate, modificationdate, suppressiondate, nbdayvaliditysurvey1, nbdayvaliditysurvey2, scenarioproductfiltertypecode, elementnomenclaturearray, analysisaxisid, analysisaxiselementarray, externaldatasourcetypecode, marketindexid, sharedscenariosourceid, shortcomment, issalesperiodautomatic, salesperiodcode, startdate, enddate, priority, budgetid, layoutid)
VALUES(2454, 5, 'Sugg. Cde Magasins', true, 45, 102, 1, 3, [299,295,304], [], [], NULL, [], '2022-03-31 17:57:51', '2022-03-31 18:00:27', NULL, NULL, NULL, NULL, [1], NULL, [], NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 410);

select * from scenario where scenarioid = 2 ;              --  select works fine for scenarioid 2 alone
select * from scenario where scenarioid in (2, 2454);   --  select works fine for scenarioid 2 and 2454 together
select * from scenario where scenarioid = 2454;         --  select get an error for scenarioid 2454 alone

thank you,

**Comment by LonwoLonwo** [comment]
Ok, thanks, I can reproduce this.

But I can see in the error message that the error comes from the Clickhouse side

```
!ENTRY org.jkiss.dbeaver.model 4 0 2022-04-13 10:52:56.444
!MESSAGE Unsupported data type: Bool
!SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2022-04-13 10:52:56.444
!MESSAGE Unsupported data type: Bool
!STACK 0
java.lang.IllegalArgumentException: Unsupported data type: Bool
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:1055)
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:882)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$MappedFunctions.deserialize(ClickHouseRowBinaryProcessor.java:459)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.readNextRow(ClickHouseRowBinaryProcessor.java:521)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:551)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:496)
	at com.clickhouse.jdbc.ClickHouseResultSet.next(ClickHouseResultSet.java:716)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.next(JDBCResultSetImpl.java:272)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:180)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.fetchQueryData(SQLQueryJob.java:751)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:575)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:451)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:458)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:847)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:3515)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:118)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:116)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:4852)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

And I don't know how we can help in this situation.

**Comment by eden0907** [comment]
Hello,Thanks you for analyzing this problem.You can comment and close this issue.I will create a new issue on the clickhouse github.Best regards,Le 13 avr. 2022 9:59 AM, Anastasiya ***@***.***> a écrit :
Ok, thanks, I can reproduce this.
But I can see in the error message that the error comes from the Clickhouse side
!ENTRY org.jkiss.dbeaver.model 4 0 2022-04-13 10:52:56.444
!MESSAGE Unsupported data type: Bool
!SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2022-04-13 10:52:56.444
!MESSAGE Unsupported data type: Bool
!STACK 0
java.lang.IllegalArgumentException: Unsupported data type: Bool
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:1055)
	at com.clickhouse.client.ClickHouseValues.newValue(ClickHouseValues.java:882)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$MappedFunctions.deserialize(ClickHouseRowBinaryProcessor.java:459)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.readNextRow(ClickHouseRowBinaryProcessor.java:521)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:551)
	at com.clickhouse.client.data.ClickHouseRowBinaryProcessor$Records.next(ClickHouseRowBinaryProcessor.java:496)
	at com.clickhouse.jdbc.ClickHouseResultSet.next(ClickHouseResultSet.java:716)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.next(JDBCResultSetImpl.java:272)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:180)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.fetchQueryData(SQLQueryJob.java:751)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:575)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:451)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:458)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:847)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:3515)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:118)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:171)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:116)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:4852)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

And I don't know how we can help in this situation.

**Comment by LonwoLonwo** [comment]
Thanks for the answer @eden0907

**Comment by eden0907** [comment]
Hello,Juste for information,Clickhouse's team has corrected this problem with jdbc driver 0.3.2 patch 7.Best regardsLe 14 avr. 2022 10:47 AM, Anastasiya ***@***.***> a écrit :
Closed #16157.
